src/python/tests/core/api/test_browser_assets.py

```diff
@@ -0,0 +1,635 @@
+        assert retval[0]['timestamp'] == self.test_data_ts_list[-1]
+
+    @pytest.mark.xfail(reason="FOGL-546, FOGL-547")
+    async def test_get_asset_sensor_readings_stats(self):
+        """
+        Verify that if more than 20 readings for an assets sensor value,
+        summary of only 20 readings are returned as the default limit
+        http://localhost:8082/foglamp/asset/TESTAPI/x/summary
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary'.format(test_data_asset_code, sensor_code_1))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == min(self.test_data_x_val_list[1:-1])
+        assert retval[sensor_code_1]['average'] == \
+               sum(self.test_data_x_val_list[1:-1])/len(self.test_data_x_val_list[1:-1])
+        assert retval[sensor_code_1]['max'] == max(self.test_data_x_val_list[1:-1])
+
+    @pytest.mark.xfail(reason="FOGL-546, FOGL-548")
+    async def test_get_asset_sensor_readings_stats_q_limit(self):
+        """
+        Verify that if more than 20 readings, limited readings summary for a sensor value are returned
+        when querying with limit
+        http://localhost:8082/foglamp/asset/TESTAPI/x/summary?limit=1
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary?limit={}'.format(test_data_asset_code, sensor_code_1, 1))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['average'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['max'] == self.test_data_x_val_list[-1]
+
+    @pytest.mark.xfail(reason="FOGL-546")
+    async def test_get_asset_sensor_readings_stats_q_sec(self):
+        """
+        Verify that if more than 20 readings, summary of only last n sec readings for a sensor value are returned when
+        seconds is passed as query parameter
+        http://localhost:8082/foglamp/asset/TESTAPI/x/summary?seconds=180
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary?seconds={}'.format(test_data_asset_code, sensor_code_1, 180))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['average'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['max'] == self.test_data_x_val_list[-1]
```

Comment:
as above! but in this make sure sum of all items in last 180 seconds

Reply:
Yes, we are doing that only. Since we have only 1 record in last 180 second, we are asserting that. We have more tests where we have more than 1 test data in last n periods and we are verifying that. Another purpose of checking with only 1 record was to assure that stats summary of 1 record is returned too and does not raises any error.